virttest/utils_misc.py

```diff
@@ -2795,33 +2812,51 @@ def format_windows_disk(session, did, mountpoint=None, size=None, fstype="ntfs")
     return False
 
 
-def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
+def format_linux_disk(session, did, mountpoint=None, size=None, fstype="ext3"):
     """
     Create a partition on disk in linux guest and format and mount it.
 
     :param session: session object to guest.
     :param did: disk serial, kname or wwn.
     :param mountpoint: mount point for the disk.
     :param fstype: filesystem type for the disk.
-    :param ostype: guest os type 'windows' or 'linux'.
     :return Boolean: disk usable or not.
     """
-    list_disk_cmd = "lsblk -p -oKNAME,MOUNTPOINT,SERIAL,SIZE,MODEL,WWN"
-    disks = session.cmd_output(list_disk_cmd)
+    disks = list_linux_guest_disks(session)
     logging.debug("Disks detail: %s" % disks)
-    for disk in disks.splitlines():
-        if did in disk:
-            kname = disk.split()[0]
-            logging.info("Create partition on disk '%s'" % kname)
-            mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
-            session.cmd(mkpart_cmd)
-            logging.info("Format partition to '%s'" % fstype)
-            format_cmd = "yes|mkfs -t %s %s" % (kname, fstype)
-            session.cmd(format_cmd)
-            logging.info("Mount the disk to '%s'" % mountpoint)
-            mount_cmd = "mount -t %s %s %s" % (fstype, kname, mountpoint)
-            session.cmd(mount_cmd)
-            return True
+    for line in disks:
+        kname = line.split('/')[2]
+        get_disk_info_cmd = "udevadm info -q property -p /sys/block/%s" % kname
+        output = session.cmd_output_safe(get_disk_info_cmd)
+        re_str = r"(?<=DEVNAME=/dev/)(.*)|(?<=ID_SERIAL=)(.*)|"
+        re_str += "(?<=ID_SERIAL_SHORT=)(.*)|(?<=ID_WWN=)(.*)"
+        did_list_group = re.finditer(re_str, output, re.M)
+        did_list = [match.group() for match in did_list_group if match]
```

Comment:
suggest you create a single function to get did_list, in order to avoid getting it many times.